### PR TITLE
installutils: Remove root password expiry when no root user is specified in imageconfig file

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

In the current images, the root password expiry gets set to the default 90 days. If the root user was configured in the imageconfig.json, it is possible to customize this expiry date. However, if no root user is specified (which is the standard case), the default root user will get the password expiry notice after 90 days and can be locked out of the system.

This change removes the root password expiry only in the case when no root user is specified in the imageconfig file.

###### Change Log  

- Remove the root password expiry only in the case when no root user is specified in the imageconfig file.

###### Does this affect the toolchain?  

NO

###### Test Methodology

- Local image testing

### CODE REVIEW

**Review by schmittjoseph — APPROVED** [review]

File: toolkit/tools/imagegen/installutils/installutils.go [hunk 1]
```diff
@@ -787,16 +787,28 @@ func addGroups(installChroot *safechroot.Chroot, groups []configuration.Group) (
...
 		ReportActionf("Adding user: %s", user.Name)
 
-		var homeDir string
+		var (
+			homeDir string
+			isRoot  bool
+		)
 
-		homeDir, err = createUserWithPassword(installChroot, user)
+		homeDir, isRoot, err = createUserWithPassword(installChroot, user)
```
Comment (PawelWMS) [inline toolkit/tools/imagegen/installutils/installutils.go:805]: Would using `rootUserAdded` in place of `isRoot` work? It feels like we're adding an unnecessary variable and an `if` statement.

File: toolkit/tools/imagegen/installutils/installutils.go [hunk 2]
```diff
@@ -814,10 +826,16 @@ func addUsers(installChroot *safechroot.Chroot, users []configuration.User) (err
 		}
 	}
 
+	// If no root entry was specified in the config file, never expire the root password
+	if !rootUserAdded {
+		err = installChroot.UnsafeRun(func() error {
```
Comment (PawelWMS) [inline toolkit/tools/imagegen/installutils/installutils.go:831]: Might be worth adding a debug log here.

**Review by PawelWMS — APPROVED** [review]

File: toolkit/tools/imagegen/installutils/installutils.go (hunk 1 above)
Comment (christopherco) [reply toolkit/tools/imagegen/installutils/installutils.go:805]: `rootUserAdded` holds whether the root user was specified in any of the imageconfig User entries. If we were to use `rootUserAdded` in place of `isRoot`, we would only track whether the last user entry was root.

File: toolkit/tools/imagegen/installutils/installutils.go (hunk 2 above)
Comment (christopherco) [reply toolkit/tools/imagegen/installutils/installutils.go:832]: Added

**Pull request merged**